### PR TITLE
Revert "[stable/fluentd] Fix missing serviceName in HPA"

### PR DESCRIPTION
**Pull request by jmthvt**
Reverts helm/charts#21202

The serviceName had to be added on the statefulset resource, not the HPA.
Since it has been done on another PR, we only need to revert this breaking change.

### CODE REVIEW

File: stable/fluentd/templates/hpa.yaml [hunk 1]
```diff
@@ -7,13 +7,12 @@ metadata:
     app: {{ template "fluentd.name" . }}
     chart: {{ template "fluentd.chart" . }}
     release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    heritage: {{ .Release.Service }}
```
Comment (Jonher937) [inline stable/fluentd/templates/hpa.yaml]:
```suggestion
    heritage: {{ .Release.Service }}
```

Remove the tab at the end please

File: stable/fluentd/templates/_helpers.tpl [hunk 2]
```diff
@@ -40,4 +40,4 @@ Create the name of the service account to use
 {{- else -}}
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
-{{- end -}}
+{{- end -}}
```
Comment (scottrigby) [inline stable/fluentd/templates/_helpers.tpl]: Should keep newline at EOF

**Review by scottrigby — CHANGES_REQUESTED** [review]
Apart from this looks good

File: stable/fluentd/templates/_helpers.tpl (hunk 2 above)
Comment (jmthvt) [reply stable/fluentd/templates/_helpers.tpl]: resolved - PTAL

**Comment by uhthomas** [comment]
Re: https://github.com/helm/charts/pull/21202

I'm also getting
```
unknown field "serviceName" in io.k8s.api.autoscaling.v2beta2.CrossVersionObjectReference
```
for v2.4.1. Could we get this merged please?

**Review by Jonher937 — APPROVED** [review]
Looks good

**Review by scottrigby — APPROVED** [review]

**Pull request merged**